### PR TITLE
Add camera bug

### PR DESCRIPTION
**Pull request by pofitlo-Git**
## About the PR

Adding a new item to the syndicate's uplink - the camera bug. a handheld device that allows you to view cameras, thereby allowing you to plan your actions much better or use them for tracking, ambushes, etc.
## Media

image
image

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

🆑
- add: Added a camera bug in the syndicate uplink that cost 4 TC and allow you watch all cameras on station.

### CODE REVIEW

**Comment by Brandon-Huu** [comment]
Does the device come out of the box with the ability to monitor the cameras?
If so, I do not think it should. The syndicate should have to manually connect it to a camera router(server) to get access to the network.

**Comment by deltanedas** [comment]
if you are going to break into ai sat or wherever the routers are you can just as easily make a camera monitor yourself (which is a t1 civ btw)

**Comment by Brandon-Huu** [comment]
> if you are going to break into ai sat or wherever the routers are you can just as easily make a camera monitor yourself (which is a t1 civ btw)

How will this function on rounds with multiple stations?
How will this affect nukies?

**Comment by TheShuEd** [comment]
I would suggest doing things a little differently. Add 3 hidden cameras, like a small item that needs to be manually placed in the plants, or slipped into a bag, and a monitor that allows you to keep an eye specifically on such cameras.

**Comment by pofitlo-Git** [comment]
> I would suggest doing things a little differently. Add 3 hidden cameras, like a small item that needs to be manually placed in the plants, or slipped into a bag, and a monitor that allows you to keep an eye specifically on such cameras.

in future. now just this. because its critically important for tactic game

**Comment by slarticodefast** [comment]
Hey, sorry for the wait on the review, we are currently quite backlogged. I opened an internal maintainer disscussion for the item balance. The code looks good on the first view and I can give it an in-game test when you fix the merge conflict.

Quick question: Did you make the item sprite yourself or did you port it from SS13? We had people misunderstand the copyright field in the meta file before, so I just want to make sure we don't violate any licenses by accident.

**Comment by pofitlo-Git** [comment]
> Hey, sorry for the wait on the review, we are currently quite backlogged. I opened an internal maintainer disscussion for the item balance. The code looks good on the first view and I can give it an in-game test when you fix the merge conflict.
> 
> Quick question: Did you make the item sprite yourself or did you port it from SS13? We had people misunderstand the copyright field in the meta file before, so I just want to make sure we don't violate any licenses by accident.

it's my sprite and i think that i can improve this a little )

**Review by slarticodefast — CHANGES_REQUESTED** [review]
Hey, I brought it up for maintainer dicussion and could you make it 3 TC and add a power cell to it for balance? Look at the handheld crew monitor to see how that is done. Inheriting from BaseHandheldComputer might be all you need to do.

**Comment by pofitlo-Git** [comment]
> Hey, I brought it up for maintainer dicussion and could you make it 3 TC and add a power cell to it for balance? Look at the handheld crew monitor to see how that is done. Inheriting from BaseHandheldComputer might be all you need to do.

I can't make it chargeable because it should be realised into the SurveillanceCameraMonitorSystem. and I literally don't sure that I should touch it considering the consequences that may be

File: Resources/Locale/en-US/store/uplink-catalog.ftl
Comment (slarticodefast) [inline Resources/Locale/en-US/store/uplink-catalog.ftl:444]:
```suggestion
uplink-cameraBug-desc = A portable device that allows you to view the station's cameras.
```

File: Resources/Prototypes/Entities/Objects/Devices/Syndicate_Gadgets/camera_bug.yml
Comment (slarticodefast) [inline Resources/Prototypes/Entities/Objects/Devices/Syndicate_Gadgets/camera_bug.yml:5]: Can you give it a proper item description for the examine window?
For example `An illegal syndicate device that allows you to hack into the station's camera network.`

**Review by slarticodefast — CHANGES_REQUESTED** [review]
I tried it out myself and it seems like SurveillanceCameraMonitorBoundUserInterface causes some problems when combined with a power cell. I guess we can include it as it is for now and adjust the balance later if necessary.
Could you fix the merge conflict and address two minor nitpicks? Should be good to merge then.

**Comment by pofitlo-Git** [comment]
> I tried it out myself and it seems like SurveillanceCameraMonitorBoundUserInterface causes some problems when combined with a power cell. I guess we can include it as it is for now and adjust the balance later if necessary. Could you fix the merge conflict and address two minor nitpicks? Should be good to merge then.

I think that It's done

**Comment by Boaz1111** [comment]
@slarticodefast shouldn't this also have the name be lowercase? not entirely sure, so I won't leave a review

File: Resources/Prototypes/Entities/Objects/Devices/Syndicate_Gadgets/camera_bug.yml
Comment (slarticodefast) [inline Resources/Prototypes/Entities/Objects/Devices/Syndicate_Gadgets/camera_bug.yml:4]:
```suggestion
  name: camera bug
```

File: Resources/Prototypes/Catalog/uplink_catalog.yml
Comment (slarticodefast) [inline Resources/Prototypes/Catalog/uplink_catalog.yml:1094]:
```suggestion
    Telecrystal: 4
```

**Review by slarticodefast — APPROVED** [review]

**Comment by slarticodefast** [comment]
I asked the other maintainers and we decided to increase the price to 4TC since it does not need a power cell now. I adjusted that number for you and will merge as soon as the tests are done. Thank you for your contribution!

**Comment by deltanedas** [comment]
tc shouldnt be relevant to powercells since you can get them all over maints and in many vendors + make microreactors anyway

**Pull request merged**
